Error decoding ReadMultipleRegistersRequest

User reported this error:
```
Caused by: java.lang.IllegalArgumentException: maxCapacity: -128 (expected:>= 0)
        at io.netty.buffer.AbstractByteBuf.<init>(AbstractByteBuf.java:67)
        atio.netty.buffer.AbstractDerivedByteBuf.<init>(AbstractDerivedByteBuf.java:31)
        atio.netty.buffer.AbstractUnpooledSlicedByteBuf.<init>(AbstractUnpooledSlicedByteBuf.java:38)
        at io.netty.buffer.UnpooledSlicedByteBuf.<init>(UnpooledSlicedByteBuf.java:24)
        at io.netty.buffer.AbstractByteBuf.slice(AbstractByteBuf.java:1188)
        at io.netty.buffer.AbstractByteBuf.readSlice(AbstractByteBuf.java:836)
        at com.digitalpetri.modbus.codec.ModbusRequestDecoder.decodeWriteMultipleRegisters(ModbusRequestDecoder.java:136)
        at com.digitalpetri.modbus.codec.ModbusRequestDecoder.decodeResponse(ModbusRequestDecoder.java:71)
        at com.digitalpetri.modbus.codec.ModbusRequestDecoder.decode(ModbusRequestDecoder.java:44)
        at com.digitalpetri.modbus.codec.ModbusTcpCodec.decode(ModbusTcpCodec.java:75)
```

This is caused by `ModbusRequestDecoder` erroneously reading a signed instead of unsigned byte here: https://github.com/digitalpetri/modbus/blob/4f8c73d97216f914746beea357b184195018d117/modbus-codec/src/main/java/com/digitalpetri/modbus/codec/ModbusRequestDecoder.java#L135

--- a/modbus-codec/src/main/java/com/digitalpetri/modbus/codec/ModbusRequestDecoder.java
+++ b/modbus-codec/src/main/java/com/digitalpetri/modbus/codec/ModbusRequestDecoder.java
@@ -132,7 +132,7 @@ public class ModbusRequestDecoder implements ModbusPduDecoder {
     private WriteMultipleRegistersRequest decodeWriteMultipleRegisters(ByteBuf buffer) {
         int address = buffer.readUnsignedShort();
         int quantity = buffer.readUnsignedShort();
-        int byteCount = buffer.readByte();
+        int byteCount = buffer.readUnsignedByte();
         ByteBuf values = buffer.readSlice(byteCount).retain();
 
         return new WriteMultipleRegistersRequest(address, quantity, values);

--- a/modbus-codec/src/test/java/com/digitalpetri/modbus/codec/ModbusRequestDecoderTest.java
+++ b/modbus-codec/src/test/java/com/digitalpetri/modbus/codec/ModbusRequestDecoderTest.java
@@ -1,0 +1,35 @@
+package com.digitalpetri.modbus.codec;
+
+import com.digitalpetri.modbus.requests.WriteMultipleRegistersRequest;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+public class ModbusRequestDecoderTest {
+
+    @Test
+    public void testDecodeWriteMultipleRegistersRequest() {
+        ModbusRequestEncoder encoder = new ModbusRequestEncoder();
+
+        WriteMultipleRegistersRequest request = new WriteMultipleRegistersRequest(
+            0,
+            64,
+            new byte[128]
+        );
+
+        ByteBuf encoded = encoder.encode(request, Unpooled.buffer());
+
+        ModbusRequestDecoder decoder = new ModbusRequestDecoder();
+
+        WriteMultipleRegistersRequest decoded = (WriteMultipleRegistersRequest) decoder.decode(encoded);
+
+        assertNotNull(decoded);
+        assertEquals(decoded.getAddress(), 0);
+        assertEquals(decoded.getQuantity(), 64);
+        assertEquals(decoded.getValues().readableBytes(), 128);
+    }
+
+}

